### PR TITLE
[13.0][MIG] account_invoice_check_total

### PR DESCRIPTION
**Pull request by jimhoefnagels**

### CODE REVIEW

**Comment by tafaRU** [comment]
Please check errors in Travis build: https://travis-ci.org/OCA/account-invoicing/jobs/591895175#L923-L943

**Comment by kittiu** [comment]
@jimhoefnagels thanks for your contribution.
This does not seem to use the pre-commit yet?
https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0#technical-method-to-migrate-a-module-from-120-to-130-branch
Would you follow that, and also fix the travis error.

**Comment by kittiu** [comment]
Also may rename this PR would be nice and easy for others to find this PR :)
```
[13.0][MIG] account_invoice_check_total
```

**Comment by kittiu** [comment]
@jimhoefnagels This has been a while. Are you still working on this?

**Review by kittiu — APPROVED** [review]

File: account_invoice_check_total/tests/test_account_invoice.py [hunk 1]
```diff
@@ -1,66 +1,79 @@
...
-
     def setUp(self):
         super(TestAccountInvoice, self).setUp()
 
         # ENVIRONEMENTS
 
-        self.account_invoice = self.env['account.invoice']
-        self.account_model = self.env['account.account']
-        self.account_invoice_line = self.env['account.invoice.line']
+        self._account_move = self.env["account.move"]
```
Comment (rven) [inline account_invoice_check_total/tests/test_account_invoice.py]: Why is this underscore needed?
```suggestion
        self.account_move = self.env["account.move"]
```

File: account_invoice_check_total/tests/test_account_invoice.py [hunk 2]
```diff
@@ -1,66 +1,79 @@
...
+        self.account_model = self.env["account.account"]
+        self._account_move_line = self.env["account.move.line"]
```
Comment (rven) [inline account_invoice_check_total/tests/test_account_invoice.py]: Why is this underscore needed?
```suggestion
        self.account_move_line = self.env["account.move.line"]
```

File: account_invoice_check_total/tests/test_account_invoice.py [hunk 3]
```diff
@@ -1,66 +1,79 @@
...
+            ],
+            limit=1,
+        )
         # Instance: Invoice Line
-        self.invoice_line = self.account_invoice_line.create(
-            {'name': 'Test invoice line',
-             'account_id': self.invoice_account.id,
-             'quantity': 1.000,
-             'price_unit': 2.99})
+        self.move_line = self._account_move_line.new(
```
Comment (rven) [inline account_invoice_check_total/tests/test_account_invoice.py]:
```suggestion
        self.move_line = self.account_move_line.new(
```

File: account_invoice_check_total/tests/test_account_invoice.py [hunk 4]
```diff
@@ -1,66 +1,79 @@
...
         # Creation of an invoice instance, wrong check_total
         # Result: UserError
-        invoice = self.account_invoice.create({
-            'partner_id': self.env.ref('base.res_partner_2').id,
-            'account_id': self.invoice_account.id,
-            'type': 'in_invoice',
-            'check_total': 1.19,
-            'invoice_line_ids': [(6, 0, [self.invoice_line.id])]
-        })
+        invoice = self._account_move.create(
```
Comment (rven) [inline account_invoice_check_total/tests/test_account_invoice.py]:
```suggestion
        invoice = self.account_move.create(
```

File: account_invoice_check_total/tests/test_account_invoice.py [hunk 5]
```diff
@@ -1,66 +1,79 @@
...
 
     def test_onchange_check_total(self):
-        invoice = self.account_invoice.create({
-            'partner_id': self.env.ref('base.res_partner_2').id,
-            'account_id': self.invoice_account.id,
-            'type': 'in_invoice',
-            'check_total': 1.19,
-            'invoice_line_ids': [(6, 0, [self.invoice_line.id])]
-        })
+        invoice = self._account_move.create(
```
Comment (rven) [inline account_invoice_check_total/tests/test_account_invoice.py]:
```suggestion
        invoice = self.account_move.create(
```

File: account_invoice_check_total/tests/test_account_invoice.py [hunk 6]
```diff
@@ -1,66 +1,79 @@
...
+            {
+                "partner_id": self.env.ref("base.res_partner_2").id,
+                # 'account_id': self.invoice_account.id,
```
Comment (rven) [inline account_invoice_check_total/tests/test_account_invoice.py]:
```suggestion
                'account_id': self.invoice_account.id,
```

File: account_invoice_check_total/tests/test_account_invoice.py [hunk 7]
```diff
@@ -1,66 +1,79 @@
...
+                "type": "in_invoice",
+                "check_total": 1.19,
+                "line_ids": [(6, 0, [self.move_line.id])],
+            }
+        )
 
-        invoice.invoice_line_ids.ensure_one()
-        invoice.invoice_line_ids.price_unit = 5.99
+        invoice.line_ids.ensure_one()
```
Comment (rven) [inline account_invoice_check_total/tests/test_account_invoice.py]:
```suggestion
        invoice.invoice_line_ids.ensure_one()
```

File: account_invoice_check_total/tests/test_account_invoice.py [hunk 8]
```diff
@@ -1,66 +1,79 @@
...
+        invoice.line_ids.price_unit = 5.99
```
Comment (rven) [inline account_invoice_check_total/tests/test_account_invoice.py]:
```suggestion
        invoice.invoice_line_ids.price_unit = 5.99
```

File: account_invoice_check_total/tests/test_account_invoice.py [hunk 9]
```diff
@@ -1,66 +1,79 @@
...
-            'check_total': 1.19,
-            'invoice_line_ids': [(6, 0, [self.invoice_line.id])]
-        })
+        invoice = self._account_move.create(
+            {
+                "partner_id": self.env.ref("base.res_partner_2").id,
+                # 'account_id': self.invoice_account.id,
+                "type": "in_invoice",
+                "check_total": 1.19,
+                "line_ids": [(6, 0, [self.move_line.id])],
```
Comment (rven) [inline account_invoice_check_total/tests/test_account_invoice.py]:
```suggestion
                "invoice_line_ids": [(6, 0, [self.move_line.id])],
```

File: account_invoice_check_total/tests/test_account_invoice.py [hunk 10]
```diff
@@ -1,66 +1,79 @@
...
```
Comment (rven) [inline account_invoice_check_total/tests/test_account_invoice.py]:
```suggestion
                "invoice_line_ids": [(6, 0, [self.move_line.id])],
```

**Review by rven — CHANGES_REQUESTED** [review]

File: account_invoice_check_total/models/account_invoice.py [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+            )
+
+    @api.onchange("check_total", "amount_total")
+    def onchange_check_total(self):
+        self.check_total_display_difference = float_round(
+            self.check_total - self.amount_total,
+            precision_rounding=self.currency_id.rounding,
+        )
+
+    def action_move_create(self):
```
Comment (kittiu) [inline account_invoice_check_total/models/account_invoice.py]: V13 no longer has this function. May be we should check on post()?

**Comment by ernestotejeda** [comment]
@jimhoefnagels , are you going to continue this PR? If you want I can continue this migration in a new PR? What do you think?

**Comment by jimhoefnagels** [comment]
@ernestotejeda  Sure go ahead

**Comment by pedrobaeza** [comment]
Superseded by #793

**Pull request closed without merging**